### PR TITLE
Added OpenMP version for map reduce - issue 204

### PR DESCRIPTION
**Pull request by Drioastorga**
Now the OpenMP version of the map-reduce is available.

### CODE REVIEW

**Comment by AndresSanchezCuadrado** [comment]
As soon as this change is introduced, we can merge it with the issue 94 and create a pull request with all the implementations tested.

File: include/omp/mapreduce.h [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+#ifndef GRPPI_MAPREDUCE_OMP_H
+#define GRPPI_MAPREDUCE_OMP_H
+
+#ifdef GRPPI_OMP
+
+#include "../reduce.h"
+
+namespace grppi{
+
+template <typename InputIt, typename Transformer, class T, typename Combiner>
```
Comment (jdgarciauc3m) [inline include/omp/mapreduce.h]: More meaningful name for type T, please.

File: include/omp/mapreduce.h [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    for(auto & map : partialOuts){
+       out = combine_op(out, map);
+    } 
+    return out;
+}
+
+
+template <typename InputIt, typename Transformer, typename Combiner>
+typename std::result_of<Combiner(
```
Comment (jdgarciauc3m) [inline include/omp/mapreduce.h]: What about using auto as return type here?

File: include/omp/mapreduce.h [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+typename std::result_of<Transformer(typename std::iterator_traits<InputIt>::value_type)>::type,
```
Comment (jdgarciauc3m) [inline include/omp/mapreduce.h]: Dow we want a version without initial value?

**Review by jdgarciauc3m — CHANGES_REQUESTED** [review]
Address comments

File: include/omp/mapreduce.h (hunk 1 above)
Comment (Drioastorga) [reply include/omp/mapreduce.h]: I have changed this typename to IdentityType

File: include/omp/mapreduce.h (hunk 2 above)
Comment (Drioastorga) [reply include/omp/mapreduce.h]: I have modified it to auto.

File: include/omp/mapreduce.h (hunk 3 above)
Comment (Drioastorga) [reply include/omp/mapreduce.h]: I think that this is part of another issue.

**Review by jdgarciauc3m — APPROVED** [review]
OK

**Comment by jdgarciauc3m** [comment]
Solves issue #204

**Pull request merged**